spring-cloud-stream-reactive/src/main/java/org/springframework/cloud/stream/reactive/StreamEmitterAnnotationBeanPostProcessor.java

```diff
@@ -0,0 +1,244 @@
+ *
+ * @author Soby Chacko
+ */
+public class StreamEmitterAnnotationBeanPostProcessor
+		implements BeanPostProcessor, InitializingBean, ApplicationContextAware, SmartLifecycle {
+
+	private final List<StreamListenerParameterAdapter<?, Object>> streamListenerParameterAdapters = new ArrayList<>();
+
+	private final List<StreamListenerResultAdapter<?, ?>> streamListenerResultAdapters = new ArrayList<>();
+
+	private ConfigurableApplicationContext applicationContext;
+
+	private Method targetMethod;
+
+	private Object targetBean;
+
+	private volatile boolean running;
+
+	@Override
+	public final void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+		this.applicationContext = (ConfigurableApplicationContext) applicationContext;
+	}
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public void afterPropertiesSet() throws Exception {
+		Map<String, StreamListenerParameterAdapter> parameterAdapterMap = BeanFactoryUtils
+				.beansOfTypeIncludingAncestors(this.applicationContext, StreamListenerParameterAdapter.class);
+		for (StreamListenerParameterAdapter parameterAdapter : parameterAdapterMap.values()) {
+			this.streamListenerParameterAdapters.add(parameterAdapter);
+		}
+		Map<String, StreamListenerResultAdapter> resultAdapterMap = BeanFactoryUtils
+				.beansOfTypeIncludingAncestors(this.applicationContext, StreamListenerResultAdapter.class);
+		this.streamListenerResultAdapters.add(new MessageChannelStreamListenerResultAdapter());
+		for (StreamListenerResultAdapter resultAdapter : resultAdapterMap.values()) {
+			this.streamListenerResultAdapters.add(resultAdapter);
+		}
+	}
+
+	@Override
+	public Object postProcessBeforeInitialization(Object bean, String beanName) throws BeansException {
+		return bean;
+	}
+
+	@Override
+	public Object postProcessAfterInitialization(final Object bean, final String beanName) throws BeansException {
+		Class<?> targetClass = AopUtils.isAopProxy(bean) ? AopUtils.getTargetClass(bean) : bean.getClass();
+		ReflectionUtils.doWithMethods(targetClass, method -> {
+			StreamEmitter streamEmitter = AnnotatedElementUtils.findMergedAnnotation(method, StreamEmitter.class);
+			if (streamEmitter != null && !method.isBridge()) {
+				targetMethod = method;
+				targetBean = bean;
+			}
+		});
+		return bean;
+	}
+
+	@Override
+	public void start() {
+		if (!running) {
+			if (targetMethod != null) {
+				Assert.isTrue(targetMethod.getAnnotation(Input.class) == null,
+						StreamEmitterErrorMessages.INPUT_ANNOTATIONS_ARE_NOT_ALLOWED);
+				String methodAnnotatedOutboundName = StreamAnnotationCommonMethodUtils.getOutboundBindingTargetName(targetMethod);
+				int outputAnnotationCount = StreamAnnotationCommonMethodUtils.outputAnnotationCount(targetMethod);
+				validateStreamEmitterMethod(targetMethod, outputAnnotationCount, methodAnnotatedOutboundName);
+				invokeSetupMethodOnToTargetChannel(targetMethod, targetBean, methodAnnotatedOutboundName);
+			}
+			this.running = true;
+		}
```

Comment:
There is a bit of a thread issue here. Again, it may be safe within its usage in AC, but I'd still go with `AtomicBoolean.compareAndSet`.
Also, not sure about the context, but is it a valid condition (for this bean to be marked as running) if IF did not materialize? Basically my concern is the lack of 'else'. It may be ok, just wanted to point it out just in case